src/test/java/bio/terra/integration/FileTest.java

```diff
@@ -192,6 +192,28 @@ public void fileParallelFailedLoadTest() throws Exception {
         assertThat("No unexpected failures", failureCount, equalTo(0));
     }
 
+    @Test
+    public void fileAlreadyExistsTest() throws Exception {
+        String gsPath = "gs://" + testConfiguration.getIngestbucket();
+        String filePath = "/foo/bar";
+
+        FileModel fileModel = dataRepoFixtures.ingestFile(
+            steward(), datasetId, profileId, gsPath + "/files/File Design Notes.pdf", filePath);
+        String fileId = fileModel.getFileId();
+        dataRepoFixtures.getFileById(steward(), datasetId, fileId);
+
+        // Re-ingesting the same file with a different load tag throws an error
+        DataRepoResponse<JobModel> launchResp = dataRepoFixtures.ingestFileLaunch(
+            steward(), datasetId, "loadTag", profileId, gsPath + "/files/File Design Notes.pdf", filePath);
+
+        DataRepoResponse<FileModel> response = dataRepoClient.waitForResponse(steward(), launchResp, FileModel.class);
+        assertThat("File already exists", response.getStatusCode(), equalTo(HttpStatus.BAD_REQUEST));
```

Comment:
I'm not sure this is a good test for the new code. The test needs to let the `IngestFileDirectoryStep` finish successfully, and then fail the step after that. This triggers the `undo` on the ingest step. If you fail within the ingest step, then its undo is not called (I believe).

Probably the only way to be sure that this is working is to try it without the fix and see if the test fails.

Reply:
So this test tries to re-ingest the same file with a different load-tag, which throws an error, and then checks that the existing file was not deleted. My understanding is that the current code would cause the existing file to get deleted, but I will also test this against develop to confirm.

It sounds like what you're suggesting is the case where a file ingest is successful but then the next step fails, triggering the undo step. I believe the unit tests cover this scenario by calling the undo step directly and then verifying whether or not the delete entry method gets called.

Reply:
Ok, so it turns out I misunderstood how flights work. On develop, if the `IngestFileDirectoryStep` raises an exception that the file already exists, it calls the `undo` of the previous step as you mentioned. This means the original file getting deleted is not an issue. In that case, I agree this test isn't particularly useful.

Reply:
I think it could be made useful, if you could force the next step to fail. You may be able to use the new stairway hooks feature to do this.

Reply:
Oh I'm not familiar with that feature! Do you have an example of how/where it gets used?

Reply:
It is the DebugInfo feature. Here are the release notes describing how the feature is used [Stairway V0.0.35 Notes](https://docs.google.com/document/d/18kiwSL1ZmOyqschJSD5FrRhGVo8heg0a7jUQuMIYbss/edit?resourcekey=0-bGQfQp_KmEzTPW5XO-dZvw#heading=h.hvefy1kh187a)

And here is the unit test for it in the Stairway code: [DebugInfoTest](https://github.com/DataBiosphere/stairway/blob/28eaca8fff4967a7149442b4111c00e9d42ebbcc/src/test/java/bio/terra/stairway/DebugInfoTest.java#L90)

Reply:
It looks like the dev branch is currently using `bio.terra:stairway:0.0.24-SNAPSHOT`. Does it make sense to update the version in this PR, or is this something that requires more coordination? I know stairway is a central component of the Data Repo so I want to double check @ddietterich @pshapiro4broad

Reply:
I would certainly recommend upgrading to 0.0.35, but it is probably @nmalfroy call.

Reply:
We should though we haven't bumped in a while (looks like we're on 0.24). Might be worth doing in a commit to see what changes are required and isolate those.  But I'm a fan of us keeping up to date with the latest and greatest stairway version

Reply:
After discussing with @ddietterich, it turns out there are several ways we could use the new `debugInfo` feature for testing so a larger discussion on how to implement that would be useful. I don't think that needs to block this PR though since there are also unit tests to confirm the behavior in this PR.